src/features/roles/role.service.ts

```diff
@@ -0,0 +1,30 @@
+import { Injectable } from '@nestjs/common';
+import { InjectRepository } from '@nestjs/typeorm';
+import { Repository } from 'typeorm';
+import { RoleEntity } from './entities/role.entity';
+
+@Injectable()
+export class RoleService {
+    constructor(
+        @InjectRepository(RoleEntity)
+        private _roleRepository: Repository<RoleEntity>,
+    ) { }
+    async findAll(): Promise<RoleEntity[]> {
+        return await this._roleRepository.find();
+    }
+    async findOne(id: number): Promise<RoleEntity | undefined> {
+        return await this._roleRepository.findOne(id);
+    }
+    async create(role: RoleEntity): Promise<RoleEntity> {
+        return await this._roleRepository.save(role);
```

Comment:
You should receive RoleModel instead of RoleEntity, then you need to map the mode into entity